UiPath.Ipc.net6.0-windows.received.txt

```diff
@@ -0,0 +1,264 @@
+        public UiPath.Ipc.Error? InnerError { get; init; }
+        public string Message { get; init; }
+        public string StackTrace { get; init; }
+        public string Type { get; init; }
+        public override string ToString() { }
+        [return: System.Diagnostics.CodeAnalysis.NotNullIfNotNull("exception")]
+        public static UiPath.Ipc.Error? FromException(System.Exception? exception) { }
+    }
+    public interface IClient
+    {
+        TCallbackInterface GetCallback<TCallbackInterface>()
+            where TCallbackInterface :  class;
+        void Impersonate(System.Action action);
+    }
+    public interface IClientState : System.IDisposable
+    {
+        System.IO.Stream? Network { get; }
+        System.Threading.Tasks.ValueTask Connect(UiPath.Ipc.IpcClient client, System.Threading.CancellationToken ct);
+        bool IsConnected();
+    }
+    public static class IOHelpers
+    {
+        public static System.IO.Pipes.PipeSecurity Allow(this System.IO.Pipes.PipeSecurity pipeSecurity, System.Security.Principal.IdentityReference sid, System.IO.Pipes.PipeAccessRights pipeAccessRights) { }
+        public static System.IO.Pipes.PipeSecurity Allow(this System.IO.Pipes.PipeSecurity pipeSecurity, System.Security.Principal.WellKnownSidType sid, System.IO.Pipes.PipeAccessRights pipeAccessRights) { }
+        public static System.IO.Pipes.PipeSecurity AllowCurrentUser(this System.IO.Pipes.PipeSecurity pipeSecurity, bool onlyNonAdmin = false) { }
+        public static System.IO.Pipes.PipeSecurity Deny(this System.IO.Pipes.PipeSecurity pipeSecurity, System.Security.Principal.IdentityReference sid, System.IO.Pipes.PipeAccessRights pipeAccessRights) { }
+        public static System.IO.Pipes.PipeSecurity Deny(this System.IO.Pipes.PipeSecurity pipeSecurity, System.Security.Principal.WellKnownSidType sid, System.IO.Pipes.PipeAccessRights pipeAccessRights) { }
+        public static System.IO.Pipes.PipeSecurity LocalOnly(this System.IO.Pipes.PipeSecurity pipeSecurity) { }
+        public static bool PipeExists(string pipeName, int timeout = 1) { }
+    }
+    public interface ISerializer
+    {
+        object? Deserialize(string json, System.Type type);
+        System.Threading.Tasks.ValueTask<T?> DeserializeAsync<T>(System.IO.Stream json, Microsoft.Extensions.Logging.ILogger? logger);
+        string Serialize(object? obj);
+        void Serialize(object? obj, System.IO.Stream stream);
+    }
+    public sealed class IpcClient
+    {
+        [System.Obsolete("Constructors of types with required members are not supported in this version of " +
+            "your compiler.", true)]
+        public IpcClient() { }
+        public UiPath.Ipc.ClientConfig Config { get; init; }
+        public UiPath.Ipc.ClientTransport Transport { get; init; }
+        public TProxy GetProxy<TProxy>()
+            where TProxy :  class { }
+    }
+    public class IpcProxy : System.Reflection.DispatchProxy, System.IDisposable
```

Comment:
Unfortunately, the number of usages is non-zero, some of which are scarier than others, i.e.: 

[`private IPackagerIpcService _remoteProxy;`](https://github.com/UiPath/Studio/blob/75909734620bc57bad2d6c1caa6f66cb3ba76b48/Robot/UiPath.RemoteDebugging.Agent/Packager/Services/PackagerExecutorIpcProxy.cs#L31C5-L31C46)

gets cast to `IpcProxy` to (un)subscribe to the underlying connection's death:
[`var ipcProxy = _remoteProxy as IpcProxy;`](https://github.com/UiPath/Studio/blob/75909734620bc57bad2d6c1caa6f66cb3ba76b48/Robot/UiPath.RemoteDebugging.Agent/Packager/Services/PackagerExecutorIpcProxy.cs#L251C13-L251C53)

We also have more legit usages around logging categories/filtering.

Reply:
for the event, we should be able to relocate the ConnectionClosed event on the IpcClient itself>
for the Dispose it might be tricker (because the IpcClient object might be lost when you call Dispose).

anyway, I'd like to try to remove this, but if it's costly, please tell me to evaluate options.